common/raw_hashtable_metadata_group.h

```diff
@@ -366,9 +366,14 @@ class MetadataGroup : public Printable<MetadataGroup> {
   auto Store(uint8_t* metadata, ssize_t index) const -> void;
 
   // Clear a byte of this group's metadata at the provided `byte_index` to the
-  // empty value. Note that this must only be called when `FastByteClear` is
-  // true -- in all other cases users of this class should arrange to clear
-  // individual bytes in the underlying array rather than using the group API.
+  // empty value.
+  //
+  // Note that this must only be called when `FastByteClear` is true -- in all
+  // other cases users of this class should arrange to clear individual bytes in
+  // the underlying array rather than using the group API. This is checked by a
+  // static_assert, and the function is templated so that it is not instantiated
+  // in the cases where it would not be valid.
+  template <bool IsCalled = true>
```

Comment:
Could this instead be written as `requires FastByteClear`, also removing the `static_assert` on `FastByteClear`?

Reply:
Can be done for the `Size == 8` too

Reply:
No, unfortunately that doesn't work. The calls to this function are within `if constexpr` blocks, which means they do get type-checked (so any requirements on the function must be satisfied). But any callees are not actually *used*, so they don't get instantiated (so `static_assert`s in their body won't be checked).

I could make the `requires` approach work by adding a wrapper function, but that seems a bit over-complicated. It also doesn't remove the need to turn the function into a template, given that only templated functions can have requirements.

Reply:
I thought if constexpr is okay for this: https://godbolt.org/z/PTbGs735E maybe there's more complex things going on though.